cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+    yield _single_qubit_gates(c1[idx_1], q_1)
+    if idx_2 == 1:
+        yield ops.CZ(q_0, q_1)
+        yield ops.Y(q_0) ** -0.5
+        yield ops.Y(q_1) ** 0.5
+        yield ops.CZ(q_0, q_1)
+        yield ops.Y(q_0) ** 0.5
+        yield ops.Y(q_1) ** -0.5
+        yield ops.CZ(q_0, q_1)
+        yield ops.Y(q_1) ** 0.5
+    elif 1 < idx_2 < 11:
+        yield ops.CZ(q_0, q_1)
+        idx_3 = int((idx_2 - 2) / 3)
+        idx_4 = (idx_2 - 2) % 3
+        yield _single_qubit_gates(s1[idx_3], q_0)
+        yield _single_qubit_gates(s1_y[idx_4], q_1)
+    elif idx_2 > 10:
+        yield ops.CZ(q_0, q_1)
+        yield ops.Y(q_0) ** 0.5
+        yield ops.X(q_1) ** -0.5
+        yield ops.CZ(q_0, q_1)
+        idx_3 = int((idx_2 - 11) / 3)
+        idx_4 = (idx_2 - 11) % 3
+        yield _single_qubit_gates(s1_y[idx_3], q_0)
+        yield _single_qubit_gates(s1_x[idx_4], q_1)
+
+
+def _single_qubit_gates(gate_seq, qubit):
+    for gate in gate_seq:
+        yield gate(qubit)
+
+
+def _single_qubit_cliffords():
+    identity = ops.X ** 0.0
+    rot_x = ops.X
+    rot_half_x = ops.X ** 0.5
+    rot_neg_half_x = ops.X ** -0.5
+    rot_y = ops.Y
+    rot_half_y = ops.Y ** 0.5
+    rot_neg_half_y = ops.Y ** -0.5
+    rot_z = ops.Z
+    rot_half_z = ops.Z ** 0.5
+    rot_neg_half_z = ops.Z ** -0.5
+
+    c1_in_xy = []
+    c1_in_xy.append([identity])
+    c1_in_xy.append([rot_x])
+    c1_in_xy.append([rot_y])
+    c1_in_xy.append([rot_y, rot_x])
+    c1_in_xy.append([rot_half_x])
+    c1_in_xy.append([rot_neg_half_x])
+    c1_in_xy.append([rot_half_y])
+    c1_in_xy.append([rot_neg_half_y])
+    c1_in_xy.append([rot_neg_half_x, rot_half_y, rot_half_x])
+    c1_in_xy.append([rot_neg_half_x, rot_neg_half_y, rot_half_x])
+    c1_in_xy.append([rot_x, rot_neg_half_y])
+    c1_in_xy.append([rot_x, rot_half_y])
+    c1_in_xy.append([rot_y, rot_half_x])
+    c1_in_xy.append([rot_y, rot_neg_half_x])
+    c1_in_xy.append([rot_half_x, rot_half_y, rot_half_x])
+    c1_in_xy.append([rot_neg_half_x, rot_half_y, rot_neg_half_x])
+    c1_in_xy.append([rot_half_y, rot_half_x])
+    c1_in_xy.append([rot_half_y, rot_neg_half_x])
+    c1_in_xy.append([rot_neg_half_y, rot_half_x])
+    c1_in_xy.append([rot_neg_half_y, rot_neg_half_x])
+    c1_in_xy.append([rot_neg_half_x, rot_neg_half_y])
+    c1_in_xy.append([rot_half_x, rot_neg_half_y])
+    c1_in_xy.append([rot_neg_half_x, rot_half_y])
+    c1_in_xy.append([rot_half_x, rot_half_y])
+
+    c1_in_xz = []
+    c1_in_xz.append([identity])
+    c1_in_xz.append([rot_x])
+    c1_in_xz.append([rot_z, rot_x])
+    c1_in_xz.append([rot_z])
+    c1_in_xz.append([rot_half_x])
+    c1_in_xz.append([rot_neg_half_x])
+    c1_in_xz.append([rot_half_x, rot_half_z, rot_neg_half_x])
+    c1_in_xz.append([rot_half_x, rot_neg_half_z, rot_neg_half_x])
+    c1_in_xz.append([rot_half_z])
+    c1_in_xz.append([rot_neg_half_z])
+    c1_in_xz.append([rot_neg_half_x, rot_neg_half_z, rot_neg_half_x])
```

Comment:
There must be a better way to generate this giant list. I think we have a way to enumerate cliffords via `cirq.ops.SingleQubitCliffordGate`. You could then do a decomposition of each into the XY gate set or XZ gate set with math.

Reply:
The API in ops.SingleQubitCliffordGate is a bit confusing to me. I have rewritten this function using some for loops and itertools.product. It is much shorter now.